### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by balerum03**
-    main.rb file should be responsible for user interaction. In this milestone you should only create content in this file.
-    Think about steps needed to play the game:
 -    what do you need from player?
 -    what do you need to display for player?
 -   Display prompts for users (hint: puts/print), collect user input (hint: get.chomp) and only mock the basic flow of the game (you can use puts or inline comments to say for example “Now, your move is displayed on the board”).
 -   Make sure that you can run your game with your executable file and play with the basic flow.
 -  The final point of this milestone is to make sure that you thought about all actions needed to play Tic Tac Toe. It does not need to work yet - this will be done in the next milestone.

### CODE REVIEW

**Review by Dandush03 — COMMENTED** [review]
# Status: 🔴 Required Changes 🔴 
### Great Job so Far! 💪 

## Let us clarify some points before you start working on this milestone again.

The game should not work, it should only pretend to be working. You can use a random number with rand to decide if the game is a winning scenario or a draw scenario
``` ruby
player_choice = gets.chomp
if move_is_valid?(player_choice)
  puts board # but don't change it, the state of the board doesn't matter for this milestone
else
  puts wrong_move_message
end
```
The program should display how the game output will look in all the different possible scenarios.
``` ruby
if winner
  puts winner_message
elsif draw
  puts draw_message
end
```

Your program is having an issue when printing the available moves, it prints it 3 times in a row.

# [__OPTIONAL__]

# try to display the available move into a board to simulate the game board before each move.

## Keep on doing this great job

# Happy Coding 🤓 💻

**Comment by bertil291utn** [comment]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow

However, in order to accomplish as the actual page, you required to do these changes, listed below:

### General requirements
- [ ] Remove `.bundle` directory form your project if you're using add it to .gitginore file
- [ ] Add a correct file name to the main one. `main.rb`

### Specific req.
- Please work on the next requirements

Basic game flow is prepared: ([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

* [ ] `[OPTIONAL]` game asks players for their names
* [ ] `[REQUIRED]` game tells which player turn it is
* [ ] `[REQUIRED]` game asks the player to select from available moves
* [ ] `[REQUIRED]` game informs player if the selected move is invalid
* [ ] `[REQUIRED]` game repeats all actions for the next player's move

- Remember all your instructions should inside a loop 

##### Implementation details

* [ ] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))
* [ ] Inside the loop, you should call actions repeated for each move
* [ ] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable:

```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

### Summary

-  Almost the entire milestone's not completed, please work on it, follow this explanation [video](https://www.loom.com/share/229a5d51bb61431594027ae65cc56d98)

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

File: bin/main [hunk 1]
```diff
@@ -1,3 +1,60 @@
...
+while i < 9
+  given_number = input_request(i)
+  display_board(board)
+  if given_number.to_i < 10
+  	puts "good job \n\n"
+    i+=1
+  else
+    while given_number.to_i > 10 || given_number.to_i <  0
+    	puts "that's not a valid option, try again \n"
+    	display_board(board)
```
Comment (bertil291utn) [inline bin/main]: ## Valid move

- Also is considered as an invalid move when the next player marks the last position again
- `[REQUIRED]` game informs the player if the selected move is invalid
  - [ ] Add the logic for this

File: bin/main [hunk 2]
```diff
@@ -1,3 +1,60 @@
...
+  puts "what's player two's name ?"
+  @name_ptwo = gets.chomp
+  system('cls') || system('clear')
+end
+
+welcome
+
+while i < 9
+  given_number = input_request(i)
+  display_board(board)
```
Comment (bertil291utn) [inline bin/main]: ## Finish game
* `[REQUIRED]` game informs player if the selected move is a winning move
*  `[REQUIRED]` game informs player if the selected move is a draw move
- Take in mind the game finish when there's a winner or draw, not necessary the player should have 9 moves
  - [ ] Replace with another variable. Mockup all the logic with these variables
  *Example:*

  ```ruby
  while theresNoWinner && !isNotDraw
  # all the game logic
  end
  ```

**Comment by bertil291utn** [comment]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your advance is amazing.

However, in order to accomplish all the requirements, you required to do these changes, listed below:
- [x] Update main file from `main` to `main.rb` 
- [x] `[REQUIRED]` game informs the player if the selected move is invalid
* [x] `[REQUIRED]` game informs player if the selected move is a winning move
* [x] `[REQUIRED]` game informs player if the selected move is a draw move

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your advance is amazing.

However, in order to accomplish all the requirements, you required to do these changes, listed below:
- [ ] Update main file from `main` to `main.rb` 
- [x] `[REQUIRED]` game informs the player if the selected move is invalid
* [x] `[REQUIRED]` game informs player if the selected move is a winning move
* [x] `[REQUIRED]` game informs player if the selected move is a draw move


### Summary

-  Remember in this milestone just mockup any variable you add, next one you have to add real logic
-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

**Comment by balerum03** [comment]
Please notice that on previous review bertil291utn mention that we had to change the name of the file from main to main.rb, it was not done since it was against the requirements.
more info at https://commandercoriander.net/blog/2013/02/16/making-a-ruby-script-executable/

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 
## Great job @team :+1: :100: :clap: , thank you for making the previous TSEs requirements

your project is approved now you can merge to development and start working on the next milestone

## Happy Coding :tophat:

**Pull request merged**